secretsstoresynccontroller/api/v1alpha1/secretstoresync_types.go

```diff
@@ -0,0 +1,144 @@
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// SecretObjectData defines the desired state of synchronized data within a Kubernetes secret object.
+type SecretObjectData struct {
+	// SecretDataValueSource is the data source value of the secret defined in the Secret Provider Class.
+	// +kubebuilder:validation:Required
+	SecretDataValueSource string `json:"secretDataValueSource"`
+
+	// SecretDataKey is the key in the Kubernetes secret's data field as described in the Kubernetes API reference:
+	// https://kubernetes.io/docs/reference/kubernetes-api/config-and-storage-resources/secret-v1/
+	// +kubebuilder:validation:Required
+	SecretDataKey string `json:"secretDataKey"`
+}
+
+// SecretObject defines the desired state of synchronized Kubernetes secret objects.
+type SecretObject struct {
+	// Type specifies the type of the Kubernetes secret object, e.g., Opaque. The controller doesn't have permissions
+	// to create a secret object with other types than the ones specified in the Enum.
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:Enum="Opaque";"kubernetes.io/basic-auth";"kubernetes.io/ssh-auth";"kubernetes.io/tls"
+	Type string `json:"type"`
+
+	// Data is a slice of SecretObjectData containing secret data source from the Secret Provider Class and the
+	// corresponding data field key used in the Kubernetes secret object.
+	// +kubebuilder:validation:Required
+	Data []SecretObjectData `json:"data"`
+
+	// Labels contains key-value pairs representing labels associated with the Kubernetes secret object.
+	// The labels are used to identify the secret object.
+	// On secret creation, the following label is added: created-by:secrets-store.sync.x-k8s.io.
+	// Creation fails if the label is specified in the SecretSync object with a different value.
+	// On secret update, if the validation admission policy is set, the controller will check if the label
+	// created-by:secrets-store.sync.x-k8s.io is present. If the label is not present, controller fails to
+	// update the secret.
+	// +optional
+	// +kubebuilder:default:={created-by:secrets-store.sync.x-k8s.io}
+	Labels map[string]string `json:"labels,omitempty"`
+
+	// Annotations contains key-value pairs representing annotations associated with the Kubernetes secret object.
+	// On secret creation, the following annotation is added: created-by:secrets-store.sync.x-k8s.io.
+	// Creation fails if the label is specified in the SecretsStore object with a different value.
+	// +optional
+	// +kubebuilder:default:={created-by:secrets-store.sync.x-k8s.io}
+	Annotations map[string]string `json:"annotations,omitempty"`
```

Comment:
As discussed, we do not need any default values here, just need to reserve our prefix.